packages/synapse-react-client/src/components/DateTimePicker/DateTimePicker.tsx

```diff
@@ -74,8 +58,8 @@ export default function DateTimePicker(props: DateTimePickerProps) {
       <MuiDateTimePicker<string | Dayjs>
         value={valueAsDayjs}
         slots={{
-          ...props.slots,
-          field: DateTimeFieldWithTzShown,
+          ...slots,
+          textField: TextFieldWithTzShown,
```

Comment:
We can just plug the custom TextField component in here.

The prior solution caused the `InputProps` passed to `TextField` to be incorrectly merged, which broke keyboard input when custom `InputProps` were specified.